src/crepe/ScriptSystem.cpp

```diff
@@ -12,4 +20,26 @@ ScriptSystem & ScriptSystem::get_instance() {
 	return instance;
 }
 
-void ScriptSystem::update() { dbg_trace(); }
+void ScriptSystem::update() {
+	using namespace std;
+	dbg_trace();
+
+	forward_list<Script *> scripts = this->get_scripts();
+	for (Script * script : scripts) script->update();
+}
+
+forward_list<Script *> ScriptSystem::get_scripts() {
+	forward_list<Script *> scripts = {};
+	ComponentManager & mgr = ComponentManager::get_instance();
+	vector<reference_wrapper<BehaviorScript>> behavior_scripts
+		= mgr.get_components_by_type<BehaviorScript>();
+
+	for (auto behavior_script_ref : behavior_scripts) {
```

Comment:
is dit geen const?

Reply:
Ik weet niet zeker wat je bedoelt met 'dit' maar in de 4 regels die ik hier zie zijn het (denk ik) geen consts? De reference naar BehaviorScript kunnen niet const zijn vanuit de component manager omdat de component instanties aanpasbaar moeten zijn, en de reference naar script kan ook niet const zijn omdat de game programmeur zelf nog niet-const velden moet kunnen aanpassen vanuit de `script.update()` functie